internals/secrethub/init.go

```diff
@@ -0,0 +1,178 @@
+	"github.com/secrethub/secrethub-go/internals/errio"
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/credentials"
+)
+
+// InitCommand configures the user's SecretHub account for use on this machine.
+type InitCommand struct {
+	backupCode                  string
+	force                       bool
+	io                          ui.IO
+	newClient                   newClientFunc
+	newClientWithoutCredentials func(credentials.Provider) (secrethub.ClientInterface, error)
+	credentialStore             CredentialConfig
+	progressPrinter             progress.Printer
+}
+
+// NewInitCommand creates a new InitCommand.
+func NewInitCommand(io ui.IO, newClient newClientFunc, newClientWithoutCredentials func(credentials.Provider) (secrethub.ClientInterface, error), credentialStore CredentialConfig) *InitCommand {
+	return &InitCommand{
+		io:                          io,
+		newClient:                   newClient,
+		newClientWithoutCredentials: newClientWithoutCredentials,
+		credentialStore:             credentialStore,
+		progressPrinter:             progress.NewPrinter(io.Stdout(), 500*time.Millisecond),
+	}
+}
+
+// Register registers the command, arguments and flags on the provided Registerer.
+func (cmd *InitCommand) Register(r command.Registerer) {
+	clause := r.Command("init", "Initialize the SecretHub client for first use on this device.")
+	clause.Flag("backup-code", "The backup code used for initializing the account on this device.").StringVar(&cmd.backupCode)
+	registerForceFlag(clause).BoolVar(&cmd.force)
+
+	command.BindAction(clause, cmd.Run)
+}
+
+type InitMode int
+
+const (
+	InitModeSignup InitMode = iota + 1
+	InitModeBackupCode
+)
+
+// Run configures the user's SecretHub account for use on this machine.
+// If an account was already configured, the user is prompted for confirmation to overwrite it.
+func (cmd *InitCommand) Run() error {
+	credentialPath := cmd.credentialStore.ConfigDir().Credential().Path()
+
+	if cmd.credentialStore.ConfigDir().Credential().Exists() && !cmd.force {
+		confirmed, err := ui.AskYesNo(
+			cmd.io,
+			fmt.Sprintf("Already found a credential at %s, do you wish the re-initialize SecretHub on this device? (this will overwrite the credential)", credentialPath),
+			ui.DefaultNo,
+		)
+		if err == ui.ErrCannotAsk {
+			return ErrLocalAccountFound
+		} else if err != nil {
+			return err
+		}
+
+		if !confirmed {
+			fmt.Fprintln(cmd.io.Stdout(), "Aborting.")
+			return nil
+		}
+	}
+
+	var mode InitMode
+	if cmd.backupCode != "" {
+		mode = InitModeBackupCode
+	}
+
+	if mode == 0 {
+		if cmd.force {
+			return ErrMissingFlags
+		}
+		fmt.Fprintf(cmd.io.Stdout(), "How do you want to initiliaze your SecretHub account on this device?\n"+
+			"1) Signup for a new account\n"+
+			"2) Use a backup code to recover an existing account\n")
+
+		option, err := ui.Ask(cmd.io, "Choose an option [1,2]: ")
```

Comment:
Shall we use `AskAndValidate` to re-prompt when something other than 1 or 2 is chosen?

Or maybe add a `ui.Choose` that does this for you (and lists the options).

Reply:
:heavy_check_mark: 

Done